### PR TITLE
[CELEBORN-1284][DOC] Add document about QuotaManager based on ConfigService

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add document about QuotaManager based on ConfigService

### Why are the changes needed?
Add document about QuotaManager based on ConfigService

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

**Review by cxzl25 — APPROVED** [review]

File: docs/quota_management.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  - hdfsBytesWritten: 10G
+  - diskFileCount: Long.MAX_VALUE
+
+The quota for tenant id `tenant_01` is
+  - diskBytesWritten: 10G
+  - diskFileCount: 1000
+  - hdfsBytesWritten: 10G
+  - diskFileCount: Long.MAX_VALUE
+
+The quota for tenant id `tenant_01` is
```
Comment (waitinfuture) [inline docs/quota_management.md:100]: for system?

File: docs/quota_management.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+to true, LifecycleManager will request master side to check whether the current user 
+has enough quota before registration of shuffle. Fallback to the default shuffle service
+of Spark when Master side checks that there is no enough quota for current user.
+
+## Quota
+Celeborn supports fine-grained quota management, now Celeborn including four indicators:
+- `celeborn.quota.tenant.diskBytesWritten`: Allowed maxed disk written bytes.
+- `celeborn.quota.tenant.diskFileCount`: Allowed maxed disk written file count.
+- `celeborn.quota.tenant.hdfsBytesWritten`: Allowed maxed hdfs written bytes.
+- `celeborn.quota.tenant.hdfsFileCount`: Allowed maxed hdfs written bytes.
```
Comment (jiaoqingbo) [inline docs/quota_management.md:37]: written bytes ->written  file count

File: docs/quota_management.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+---
+
+Quota Management
+===
+
+The Celeborn allows the administrator to set quotas for the number of names used.
```
Comment (pan3793) [inline docs/quota_management.md:22]:
```suggestion
Celeborn allows the administrator to set quotas for the number of names used.
```

File: docs/quota_management.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+When Master side sets `celeborn.quota.enabled` to true, the master will enable
+to check the quota via QuotaManager. When Client side also sets `celeborn.quota.enabled`
+to true, LifecycleManager will request master side to check whether the current user 
+has enough quota before registration of shuffle. Fallback to the default shuffle service
+of Spark when Master side checks that there is no enough quota for current user.
+
+## Quota
+Celeborn supports fine-grained quota management, now Celeborn including four indicators:
+- `celeborn.quota.tenant.diskBytesWritten`: Allowed maxed disk written bytes.
```
Comment (RexXiong) [inline docs/quota_management.md:34]:
```suggestion
- `celeborn.quota.tenant.diskBytesWritten`: Maximum allowed disk write bytes.
```

File: docs/quota_management.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  - hdfsBytesWritten: 10G
+  - diskFileCount: Long.MAX_VALUE
+
+The quota for tenant id `tenant_01` is
+  - diskBytesWritten: 1G
+  - diskFileCount: 100
+  - hdfsBytesWritten: 1G
+  - diskFileCount: Long.MAX_VALUE 
+
+### DataBase Store Backend
```
Comment (SteNicholas) [inline docs/quota_management.md:106]:
```suggestion
### Database Store Backend
```

File: docs/quota_management.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+DataBase Store Backend will read [quota configuration](#Quota) from user specified database.
```
Comment (SteNicholas) [inline docs/quota_management.md:107]:
```suggestion
Database Store Backend will read [quota configuration](#Quota) from user specified database.
```

File: docs/quota_management.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+## QuotaManager
+The Celeborn will initialize a QuotaManager in the master side. Currently, QuotaManager supports three level configuration: 
+  - Tenant User Level: When the tenant user level quota config is null or empty, fallback to the tenant level quota config. When the tenant level config is null or empty, fallback to the system level config again.
+  - Tenant Level: When the tenant level config is null or empty, fallback to the system level config.
+  - System Level: When the system level config is also null or empty, fallback to default value `Long.MAX_VALUE`.
+
+QuotaManager use dynamic [config service](#) to store quota settings mentioned in [Quota](#Quota),
+QuotaManger also support two types of store backend configured by parameter `celeborn.dynamicConfig.store.backend`:
+  - FS: [FileSystem Store Backend](#FileSystem Store Backend)
+  - DB: [DataBase Store Backend](#DataBase Store Backend)
```
Comment (SteNicholas) [inline docs/quota_management.md:59]:
```suggestion
  - DB: [Database Store Backend](#DataBase Store Backend)
```

File: docs/quota_management.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+The quota for tenant id `tenant_01` is
+  - diskBytesWritten: 1G
+  - diskFileCount: 100
+  - hdfsBytesWritten: 1G
+  - diskFileCount: Long.MAX_VALUE 
+
+### DataBase Store Backend
+DataBase Store Backend will read [quota configuration](#Quota) from user specified database.
+For how to use DataBase store backend can refer to [config service](./config_service.md).
```
Comment (SteNicholas) [inline docs/quota_management.md:108]:
```suggestion
For how to use database store backend can refer to [config service](./config_service.md).
```

File: docs/quota_management.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+## User Identifier
+The Celeborn supports user identifier for two level:
+  - tenant id
+  - username
+The client LifecycleManager will request master side to check quota for current user that defined by user setting.
+User can set `celeborn.quota.identity.provider` to choose a identity provider, now we support two type provider:
+  - `org.apache.celeborn.common.identity.HadoopBasedIdentityProvider`: username will be obtained by `UserGroupInformation.getUserName()`, tenant id will be default.
+  - `org.apache.celeborn.common.identity.DefaultIdentityProvider`: username and tenant id are default values or user-specific values setting by `celeborn.quota.identity.user-specific.tenant` and `celeborn.quota.identity.user-specific.userName`.
+Celeborn use `org.apache.celeborn.common.identity.DefaultIdentityProvider` as default.
```
Comment (SteNicholas) [inline docs/quota_management.md:47]:
```suggestion
Celeborn uses `org.apache.celeborn.common.identity.DefaultIdentityProvider` as default. 
```

File: docs/quota_management.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+Celeborn use `org.apache.celeborn.common.identity.DefaultIdentityProvider` as default. 
+Also, the Celeborn user can implement their own specified identify provider inheriting interface `org.apache.celeborn.common.identity.IdentityProvider` by yourself.
+
+## QuotaManager
+The Celeborn will initialize a QuotaManager in the master side. Currently, QuotaManager supports three level configuration: 
+  - Tenant User Level: When the tenant user level quota config is null or empty, fallback to the tenant level quota config. When the tenant level config is null or empty, fallback to the system level config again.
+  - Tenant Level: When the tenant level config is null or empty, fallback to the system level config.
+  - System Level: When the system level config is also null or empty, fallback to default value `Long.MAX_VALUE`.
+
+QuotaManager use dynamic [config service](#) to store quota settings mentioned in [Quota](#Quota),
```
Comment (SteNicholas) [inline docs/quota_management.md:56]:
```suggestion
QuotaManagers use dynamic [config service](#) to store quota settings mentioned in [Quota](#Quota),
```

File: docs/quota_management.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+QuotaManger also support two types of store backend configured by parameter `celeborn.dynamicConfig.store.backend`:
```
Comment (SteNicholas) [inline docs/quota_management.md:57]:
```suggestion
QuotaManger also supports two types of store backend configured by parameter `celeborn.dynamicConfig.store.backend`:
```

**Comment by SteNicholas** [comment]
@AngersZhuuuu, the quota management in developer docs shoud only introduce the feature itself. About how to use dynamic config to manage quota should be introduced in user guide. Therefore, we could add user guide in navigation.

File: docs/quota_management.md (hunk 1 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:100]: Done

File: docs/quota_management.md (hunk 4 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:34]: Done

File: docs/quota_management.md (hunk 3 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:22]: Done

File: docs/quota_management.md (hunk 5 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:106]: Done

File: docs/quota_management.md (hunk 10 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

File: docs/quota_management.md (hunk 8 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

File: docs/quota_management.md (hunk 7 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:59]: Done

**Comment by AngersZhuuuu** [comment]
ping @SteNicholas @RexXiong Changed after `configuration.md`, please take a look

File: docs/quota_management.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+doesn't set a specified a named quota.
+
+When Master side sets `celeborn.quota.enabled` to true, the master will enable
+to check the quota via QuotaManager. When Client side also sets `celeborn.quota.enabled`
+to true, LifecycleManager will request master side to check whether the current user identifier
+has enough quota before registration of shuffle. Fallback to the default shuffle service
+of Spark when Master side checks that there is no enough quota for current user.
+
+## Quota
+Celeborn supports fine-grained quota management, now Celeborn including four indicators:
```
Comment (SteNicholas) [inline docs/quota_management.md:33]: [Quota](https://celeborn.apache.org/docs/latest/configuration/#quota) introduces the quota configs. Therefore this only adds the link of quota config.

File: docs/quota_management.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+---
+
+Quota Management
+===
+
+Celeborn allows the administrator to set quotas for the number of names used.
+The administrator also can set a system level default quota for all user that 
+doesn't set a specified a named quota.
+
+When Master side sets `celeborn.quota.enabled` to true, the master will enable
```
Comment (SteNicholas) [inline docs/quota_management.md:26]:
```suggestion
- When the `Master` side sets `celeborn.quota.enabled` to true, it enables quota checks through the `QuotaManager`.
- If the `Client` side also sets `celeborn.quota.enabled` to true, the `LifecycleManager` will request the `Master` side to check if the current user identifier has enough quota before registering the shuffle.
- If there is not enough quota for the current user, the `Master` side will initiate a fallback to the default shuffle service of Spark.
```

File: docs/quota_management.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+---
+
+Quota Management
+===
+
+Celeborn allows the administrator to set quotas for the number of names used.
```
Comment (SteNicholas) [inline docs/quota_management.md:22]:
```suggestion
Celeborn offers flexibility to administrators by allowing them to set quotas for individual users as well as providing a system-level default quota for those users who do not have a specified named quota. This ensures a level of control and customization in managing the quotas for the system.
```

File: docs/quota_management.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+of Spark when Master side checks that there is no enough quota for current user.
+
+## Quota
+Celeborn supports fine-grained quota management, now Celeborn including four indicators:
+- `celeborn.quota.tenant.diskBytesWritten`: Maximum allowed disk write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.diskFileCount`: Maximum allowed disk write file num, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsBytesWritten`: Maximum allowed hdfs write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsFileCount`: Maximum allowed hdfs write file num, default value `Long.MAX_VALUE`.
+
+## User Identifier
```
Comment (SteNicholas) [inline docs/quota_management.md:39]: Remove this section.

File: docs/quota_management.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+User can set `celeborn.quota.identity.provider` to choose a identity provider, now we support two type provider:
+  - `org.apache.celeborn.common.identity.HadoopBasedIdentityProvider`: username will be obtained by `UserGroupInformation.getUserName()`, tenant id will be default.
+  - `org.apache.celeborn.common.identity.DefaultIdentityProvider`: username and tenant id are default values or user-specific values setting by `celeborn.quota.identity.user-specific.tenant` and `celeborn.quota.identity.user-specific.userName`.
+Celeborn uses `org.apache.celeborn.common.identity.DefaultIdentityProvider` as default. 
+Also, the Celeborn users can implement their own specified identify provider inheriting interface `org.apache.celeborn.common.identity.IdentityProvider` by yourself.
+
+## QuotaManager
+Celeborn will initialize a QuotaManager in the master side checking quota.
+QuotaManager uses [dynamic config service](developers/configuration.md#dynamic-configuration) to store quota settings mentioned in [Quota](#Quota).
+QuotaManger also supports two types of store backend configured by parameter `celeborn.dynamicConfig.store.backend` to store quota setting:
```
Comment (SteNicholas) [inline docs/quota_management.md:53]: Remove this introduction of store backend for dynamic config.

File: docs/quota_management.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+  - FS: [FileSystem Store Backend](#FileSystem Store Backend)
+  - DB: [Database Store Backend](#Database Store Backend)
+
+Currently, QuotaManager supports [three level](developers/configuration.md#Config Level) quota settings.
```
Comment (SteNicholas) [inline docs/quota_management.md:57]: Remove this introduction.

File: docs/quota_management.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+
+### FileSystem Store Backend
```
Comment (SteNicholas) [inline docs/quota_management.md:59]: We could introduce an example for quota management and provide how to config quota for this example.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: docs/quota_management.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+limitations under the License.
+---
+
+Quota Management
+===
+
+Celeborn allows administrators to set quotas for the number of names used.
+A system-level default quota can also be set for all users who haven't specified a named quota.
+
+When `celeborn.quota.enabled` is set to true, the master checks the quota via QuotaManager.
```
Comment (SteNicholas) [inline docs/quota_management.md:25]:
```suggestion
When `celeborn.quota.enabled` is set to true, the master checks the quota via `QuotaManager`.
```

File: docs/quota_management.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+If the same setting is enabled on the client side, the LifecycleManager will request
```
Comment (SteNicholas) [inline docs/quota_management.md:26]:
```suggestion
If the same setting is enabled on the client side, the `LifecycleManager` will request
```

File: docs/quota_management.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+the master to check if the current user identifier has enough quota before registration of shuffle.
+If there's not enough quota, the system will fallback to the default shuffle service of Spark.
+
+## Quota Indicators
+
+Celeborn supports fine-grained quota management, including four indicators:
+
+- `celeborn.quota.tenant.diskBytesWritten`: Maximum allowed disk write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.diskFileCount`: Maximum allowed disk write file num, default value `Long.MAX_VALUE`.
```
Comment (SteNicholas) [inline docs/quota_management.md:35]:
```suggestion
- `celeborn.quota.tenant.diskFileCount`: Maximum allowed number of disk write files, of which default value is `Long.MAX_VALUE`.
```

File: docs/quota_management.md [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+- `celeborn.quota.tenant.hdfsBytesWritten`: Maximum allowed HDFS write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsFileCount`: Maximum allowed HDFS write file num, default value `Long.MAX_VALUE`.
```
Comment (SteNicholas) [inline docs/quota_management.md:37]: Ditto.

File: docs/quota_management.md [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+
+## User Identifier
+
+Celeborn supports user identifiers on two levels:
+
+- Tenant ID
+- Username
+
+The client's LifecycleManager will request the master to check the quota for the current user defined by user setting. Users can set `celeborn.quota.identity.provider` to choose an identity provider. We currently support two types:
```
Comment (SteNicholas) [inline docs/quota_management.md:46]:
```suggestion
The `LifecycleManager` will request the `Master` to check the quota for the current user defined by user setting. Users can set `celeborn.quota.identity.provider` to choose an identity provider. Celeborn support the following types at present:
```

File: docs/quota_management.md [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+Celeborn supports fine-grained quota management, including four indicators:
+
+- `celeborn.quota.tenant.diskBytesWritten`: Maximum allowed disk write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.diskFileCount`: Maximum allowed disk write file num, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsBytesWritten`: Maximum allowed HDFS write bytes, default value `Long.MAX_VALUE`.
+- `celeborn.quota.tenant.hdfsFileCount`: Maximum allowed HDFS write file num, default value `Long.MAX_VALUE`.
+
+## User Identifier
+
+Celeborn supports user identifiers on two levels:
```
Comment (SteNicholas) [inline docs/quota_management.md:41]: Remove this introduction.

File: docs/quota_management.md (hunk 14 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

File: docs/quota_management.md (hunk 17 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

File: docs/quota_management.md (hunk 16 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

File: docs/quota_management.md (hunk 19 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:25]: Done

File: docs/quota_management.md (hunk 21 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:35]: Done

File: docs/quota_management.md (hunk 23 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md:46]: Done

File: docs/quota_management.md (hunk 24 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: DOne

File: docs/quota_management.md (hunk 20 above)
Comment (AngersZhuuuu) [reply docs/quota_management.md]: Done

**Review by SteNicholas — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]
LGTM, thanks. Merge to main(v0.5.0)

**Pull request closed without merging**